src/core/account_manager.mjs

```diff
@@ -515,4 +540,40 @@ export class AccountManager {
       throw new FullstackTestingError(errorMessage, e)
     }
   }
+
+  /**
+   * Fetch and prepare address book as a base64 string
+   * @param nodeClient node client
+   * @return {Promise<string>}
+   */
+  async prepareAddressBookBase64 (nodeClient) {
+    // fetch AddressBook
+    const fileQuery = new FileContentsQuery().setFileId(FileId.ADDRESS_BOOK)
+    let addressBookBytes = await fileQuery.execute(nodeClient)
+
+    // ensure serviceEndpoint.ipAddressV4 value for all nodes in the addressBook is a 4 bytes array instead of string
+    // See: https://github.com/hashgraph/hedera-protobufs/blob/main/services/basic_types.proto#L1309
+    const addressBook = HashgraphProto.proto.NodeAddressBook.decode(addressBookBytes)
+    let modified = false
+    for (const nodeAddress of addressBook.nodeAddress) {
+      // overwrite ipAddressV4 as 4 bytes array if required
+      if (nodeAddress.serviceEndpoint[0].ipAddressV4.byteLength !== 4) {
+        const ipAddress = nodeAddress.serviceEndpoint[0].ipAddressV4.toString()
+        const parts = ipAddress.split('.')
+        if (parts.length !== 4) {
+          throw new FullstackTestingError(`expected node IP address to have 4 parts, found ${parts.length}: ${ipAddress}`)
+        }
+
+        nodeAddress.serviceEndpoint[0].ipAddressV4 = Uint8Array.from(parts)
+        modified = true
+      }
+    }
+
+    if (modified) {
+      addressBookBytes = HashgraphProto.proto.NodeAddressBook.encode(addressBook).finish()
+    }
+
+    // convert addressBook into base64
+    return Base64.encode(addressBookBytes)
```

Comment:
I remember writing this function, but I did not test whether `Base64.encode` works when `addressBookBytes` is not modified. 

`addressBookBytes` is Uint8 array when fetched from the network, but then `HashgraphProto.proto.NodeAddressBook.encode` generate a buffer.

Reply:
sure, I'll add a test

Reply:
I want to do a mock test, but I want this to merge before I run my release.  I did this test to verify your code:
![image](https://github.com/hashgraph/solo/assets/5412850/2fe68de2-33ce-47fc-9e67-1cb5f9338e0c)

Here is a new Issue to track writing the test: https://github.com/hashgraph/solo/issues/63